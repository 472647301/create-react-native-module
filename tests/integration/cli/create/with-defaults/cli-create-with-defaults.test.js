const execa = require('execa');
const path = require('path');

const fs = require('fs-extra');

const tar = require('tar');

test('CLI creates correct package artifacts on file system, with no options', async () => {
  // remove test artifacts just in case:
  fs.removeSync(`check-integration-test-package.tar`);
  fs.removeSync(`react-native-integration-test-package`);

  await execa.command(
    `node ${path.resolve('bin/cli.js')} integration-test-package`);

  // uses tar output to check both correct tree and correct contents
  // (using `gzip: true` to make the output less redundant)
  await tar.c({
    file: `check-integration-test-package.tar`,
    gzip: true,
    portable: true,
    noMtime: true
  }, ['react-native-integration-test-package']);

  const checkObject = fs.readFileSync(`check-integration-test-package.tar`);

  // TBD THE OUTPUT OF THIS SNAPSHOT SEEMS TO BE TOO VERBOSE:
  // expect(checkObject).toMatchSnapshot();

  // less verbose text output, seems to be just under 9K:
  const base64 = checkObject.toString('base64');
  expect(base64).toMatchSnapshot();

  // cleanup generated test artifacts:
  fs.removeSync(`check-integration-test-package.tar`);
  fs.removeSync(`react-native-integration-test-package`);
});
