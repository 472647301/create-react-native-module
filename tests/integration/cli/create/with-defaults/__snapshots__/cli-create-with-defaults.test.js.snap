// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CLI creates correct package artifacts on file system, with no options 1`] = `"H4sIAAAAAAAAE+0963rayJLnr/UUPSR7DI4tbjZ27DgzMsgOGQx8XDzxyeSAgDYoFhIrCcdeH7/Jfvtw+yRb1d0SEggDjuOZ2UN/+WLorq7urq5bV1+wqdZ1d0zN1W/ojm66tG/DZ8vccanj7oy07rXWp8m/fVdKpVL7e3skKl+kcFkqvZvaz5G/7X1fs8ulseNqNnTle/F4A4ka0J842UvNv9zXXc11bb0zhvxV2wB65HZ3SVQ+S5mZ+c/t59JAyqcOapX0bz7/W/Koczuyra8EZvzWlf7o/qzTy6bl5V/vm5ZNn9LGAvnPZadsA8h/aj+zlv+XSK9Ipf5JeiXJhXqr7sIES9IrYlo9Kn91IBs/tYZWb2xQJymZo+FOj3bGfdmw+tKdZps71LYtm32Fep+6AA61OmPd6CUlplvGDrWln3r0Shsbrp+xJQNSmr7JToq8DF6UmS7KsKIRtZ0R7SK3BgHC2bddbKOnuRrUuO12B7R7bY1dhvmG9nY0R4duFqgN4L0CBxsMtRH80Ue8DiIAxnCphLoRMEPeN8u+dkAiGIkUs2dbei9ZBJExDP3jZNQyINckGeSoZ1DJsLqaIQMS6KKrUwfbGBqI4aSZ/xXqdK93oG9J6XcZPzOqXdM7h03FC8z/kvIPM//D5D/C/8uBUljL/0uk5ea/piqFc1Ue9p7UxuPzn03tpsNlqfRedm89/y+SXpGlOAAU1ityRl1XN/sEKebSniS1XxNQDEQ3IccwlsNEdnYc7Ya2EeMrcm45rnFHtLFrDQG26+Fi9VgDQaTE0M3r5Zrx8GvmWDOmsGLJK6JX6pKUlknRJJ/yYGC2AYq4A0qExiemdqP3NVDE28TW+wOXdA29e03aJb1jazbo8jb53//5b9JWej1yqoOBJK5FPt9ZY9tDsekAkiH90pYyMjmzsLwdtKgCwXIDIprZIxq01S5OQBoAUeUAYKEANbbclrLLDcuhBuYEuywTHE7b0DtzWtHaOIypUbZP0PaR6kBz/FGVcK5OdFOz78hvujsgE7pJuzKpjc0QFhJv54e9N7V24p2YIGFi2SRVRtQk4xFpa8LuaqNR0rG7yaGmm8mv2o2W/CzL8pfkOXxXRiOYKdZzGYvaEiE7fFz6cGTZMJPWUGZUN1if7uTowYo/R2zISENe3SGay7661ohYV+zjFTDApBmTfiOPoownfJyG7rjA0+7YNmmPdO5YZrtPvRoOwg6pO7B6yEYwOAqMwNq0DMP6hhIJcsHZz6ePw2XVEX5I+xA6t9FuIyk2dLNrjHuUbB4uxXmbWEfMUnzZOglZ1CjoNjkmSBCUkbhtWW5VsNoEYptsbMpyMuRtLmeZxHg3E974GOuD9+ZGkggUAbhnrAhkBQhJzS5wJOmAn3aNkhJiMObRRVAQE7DQCAZEViaMh4ap1KaDyhW+Ip86XVsfuZLg0Wj+IVe2NSSby7V1JEnJJGlUCpVD8tsAmdYiPYt8Q2lEEnBS/yxFN3WE/VoHBP5/p5Wk7Ilx4NXjv3t76XX890XScvOvg6K8lb+uHPnlacH6L5Oe9f/3MCa09v9/fBLG5p6UGQucc+NLHiLsDJqTrgWeNEDPMU4PYOpDiKAKvWUtiFDNnJpHazvzx6Ql5d9yvmMP8An7f7n91Fr/v0Rabv7FX7AAlrl6G4/r//Redqoslc5lc/tr/f8S6R6WIzEMUsQOSWwpZohtYxVXdw1Wp4Z1hNIPKneC2p1Ug3VuqO1ACdZKyyk5xXN7lK97RAmuVngBLu0xx3M+eC4HdqDgnq3EYtg3BKPdgUV+j6m4IXFITFhcYwdwW0C/0mnv9xj5+98JvdVdko5BzQeGzaYjy9Fdy74LILwbsZH1dZc1CVlj2xA5bwauO3IOk0n4PBh3ZFgGio8trdu1xqa73MJVDmDvaA5t8haeCftkgNf07ptl95Ben3lrQQQI94XBaWN3YNkTIngscYkhmjJ+EZ2lMC2sqxi8QahfWA52ddKqoXcpLMIR7LzYiAXzMArgIS8V82q5rsbEVGi9YajYjzpzgBGldiGwaJ90lg0Ja/wznZMP5LTX19BQofj9cUreeyunduyunCLvkAlvJ53u0Zul0Gfnof8nw55OMZTSw1/CpVlO/y/H0iOrh+I208YC/Z9LTev/zO5+Zq3/XyTZ9D/Huk1JDE17TJK8KP0x+VivlOWRZjs0jkKJ8dIe//TV0s14q9XT7VZrG+Qy4BzEEomEJFWt3uFhHVhBxrhfzyL/cv4FMuHIKNnES8fLGhxWVRgPr6oo++wblS8MyhkPhxhwnoYKGhkOGchByHfvdgpqPS+RmbTc/shsPSLwOfLAGtIRElWM+hmVvCMLteqTFPUtFLwKFx3Dmu0QDRs5fi90MjnEmDX7flosqS1PGYPqQrzcIDge3vugKWCVIvS/qDoyNPfKsoeO3zKsH1idt2j0oSM34nsavQBRywF8XY87WC0gCoN6VpMLrWt9hvbVvc9VD9iJQC9aV2w/B8iJK5+treSWfD/YHm473/Qr94ETXkiO09LsLkC69phKgrOEEbkT3lFM2hAzEiySZTkmwTfpD7UTq8X/nrYPvED/Z7LZcBme/9vfXev/l0h8RqVjliSpeIWbcuSbZrLNgtG4Y+jOQOxTdYjmEI0MtRtqTrZP7rbFLgtCgbZxXDpySIeCBqAeAtyA0dgmkKfGAbk5Gh6yzb0TqDS2cTuNDAA3a01s/pHP9cKvX+KeBgAXjRp4lEYWHMmUgeOOe7olgpQDd2gk2F7p5/LiqmbvOtkf62AMQtXFfjFlO27T3dNIe/pUT5vt//GdVt1BevSoDZ81l4zAWLqOt9nnDQtGxfoIXWS7LNAPGezpcbKJB4+S93j+CJXrQ5Lvmt4lvQNHDvQYoHc6YxPPFzmrVJTEBlmBGtTldG6z2WyLLnsbs21nDFZbTvKtr28ePRS7OwA94bSlPZlcUFu/uuODBIXPFlvUxf3QPjUpaBHaI1yN6o63B81ZR5DH34bqWraNO8Aeb5jjYQf68keLxr9FWk3/BzdEl2/jcf2/n8NYTyg/g1Ghtf5/icRmlHvCYrlLb13Z0a6oeuuegTyDJ4Z6DpQ8KIAO8APZee97u8HdfKw30Jw4QifIzzNlfeqKskMfF0P0IHE3W0RicDv+3m+hb1l9g8YTfsZX8Gldaosc5rWR8F7+pHIySc41twtmCdBjPeLSIXqnYhs9FLuKp+RcKhGsG+F3Xmld2rGs65DHmewYVieJ1XeAmToGTXrNRArOq1I65zfTNTTHGWnuIB7DQymebXIty3BkVuuQ1zp8fT+Zlfgmz6wa475uXnC9ublNNrPyrpzeTDzEPPIAgbTRyLgjIwZ6SDaDzYjjL5vTMExPb7LTAx4BbzRjzA5QLEFKjIOQJEGKSE+g497bJem4t1N6+5QGlp4o1kCPXsFa7lRplhqtfOW8ioslsN+tC7VWL1bKICGZgxDQSbNYKrQalUqpHgCKZQ7klJyNhUDPi+UpXMAbQYCGUjtTG7PtSaLDjNvFWZR671qwAgmyykwpMMoj40HOYVRoIA9GIZwpDSCMGHsCZVRs/+Ut80rvCxEd6uacLodKAtinyMWZHPQ36JY5mKYLA8hmScvxCTcED8+RdDAH174sdM0DfATPFLkVFkLwtI7WsWy3YrIgNGo5h4pYoBSh37g3FFJXimGgBxVWTB/r26TS+bqTJ3x16mz7fmQHz9cBxgR3soTXykUS/OunacKg+H6HOlxajLPf1czSwgxS4zUztg0Sez05f5Z87PCZhy4WsFZs6vJAR1szwBTB/EaYIB26Q4cAxDdEmIL3hsKPH4bOjE1peFbESwIK/g0qd97eFcg9ShOsTdj08dmrWsM4lo2sYUL0g33lntxHBzrCj+P1wUDf3LGonYzZdWNsj9Cw8qBfA2x2HP33OJ7MC8b4NhMyXhVLcNML7XgH+QKszAJ9gTZltl/kF2uwagJKuMVeCAhr+TDeauA4BCJyJ/6BbY1HqF2nj3TGfJBgpC+IKpAfYowgzGR/SBa7NJIPK8JrQYcjkD2VG0kUAToDuLgX5A3Z5AIw1GDBbSc3IScCtbedMtNET3f4dU6kyiY2sBmCeZAmnwKUFGvo6TH7BRGj1sNzzOOKsrdOXUwkUWEGONhD/j/6OldADBV9FVRt9/7hYvBZha1wromQ54/ajdazunEMix4S8S0RGIGISB57FURwsE5dR8bNSXa4WHbsLmgQZ9ajI2+O+QI47tUHwXfzXnHisRreaWywXXkh5EyJOCCekHdyh7bIN+0gkZpTghmNJyZY/VO+LHyJXd0MuswRlPio2T4x7G3hVDsVcAjDcEEasb7jBiue8t38ygEmrMSMS7g2yp2rm2w4QLh5fapzUxfq0/yGhWGMbni5eVtArymX+QLMrmYCQrz4cA/eMfsaXBuh1mXMfOyVMh6Wu9pIBzOt/1dwVcNGHvuq2a/vWfQmNmciPEzYvTyf/MSUzLGRR8BF0T0oPXyUQjEHBVwTUZ/ZmZkLMmGomemdCiWJcTZHhqX1QvMbZHugYeg7kF3UnyxFA/6VzMxzgQLSuxmdBC4GL8GYXCj8AUrMdHc6tt7r0yRDsY0AvbtQKNQKOVahpu9Qcx+SGMoweDCv7ycexgO6GJ5fwnDHpBCSObYc7WuEonu29f9q8R+8cbF6G6uf/8plU3vr818vkZY//zfnPtBgcRsLzv+m96bnP5Peza73f14kvRIHgN8xtZOs5RsnTP/x87vyAPymX5AtbFy4zDv2e0jKdVigoqP1bgoD1sdNzj96nOsUnb5b/oeL21gg/7nc9P4vyP/e+v2HF0me/MfmqXdwU0ADhCMZ0bCSBLLfUj9VK7VG67xSaJZU8HBDmWrjQ6UQdzREd86uNB7Gy/U6rETNPtlKOOyDYvfH2JbYB6xqNnjEoIEO46ZlmmPwt8v1MiuCKhzGr9IVGwuHcWi2Rp0R+Irg+Js9ap/gFb+EB5CQuFfo348rekOEZR847SAVYwAFj86hV2ODXI3NLo4XHHf3Ttrw0MR/+fzZHwDvPt51PbXsoeYekl9i4f4dkv/4hYRHiVmxbRKG254C+vIlcYTL2+dXpd8t//6t4/mO4cr+XyabQZ2w9v9+fHqG+fffRZnHAU+Y/1x6fz3/L5FWX/+xG/crtbH6+m8/k02v5/8l0jPqfy9mKR6Um7TxqP+XSef2Z85/57J76/f/XiSB//PT661m4/Rg6/VP4BNtiMDahb/7kj4CdweDrexALEA8QIbFVnsToN2cn8mhpOQWOaF93STVk0/sdRDciiAOZV4U2UpKGxsnWXX/ZO9ASefzGSWfyqVSCnBKpkCg7rzVBh5AEAFIQIJN6Y4Gf4KtHLEgeo1eQb5o4+1KbWwljwiMEkBUszd3AKEh5q3RHXsEhoGyh1DCg907ONlLpwtv08pbNZc/2Eul0rv7uf0cdsSvLIYE4Bv+sCIwHyEA29xUWAvnGFg9JrBs2t89yOZ29xlAD4fkDtgpZh7bTr6OV2vgmINHXlbO1UTMg6uPO/xQHh7aZ4cQWIl3CjqOXxIsyx6bFdO4AweXB1HRO61ajgsSD1PioBt8TFIIGiLgAvKESHnKJ4/a1OxOETGd3T1I72ZSaUXZVZVcNpU62c+Ds4BEfOTJmmlOCbVwROgtvheju5jdYGfkvYC2rNlH3sZA0SziCVE2PDLihH2kzSOxfdSwKSLEcxGNlqB+vVUo1hiL+UJwsAKDDhaMB6dNNUEv89nYPSIgvu6vpvXNDAyRdw+1t9yVB/6Q5rU5NZ7YO7bz+j4WHsaqcvbMwwAN1F00kuGjI5mw7Fw2DDMrLk6ZB7xA8A+iBH9SO0ryo3AvJ/rPJ7iLxhcixhnbjI+S2DSX2P1CUGKrttUbo8GIGDtDxToOgmj0YBK84XyHEtiekEPszHldYHlz2IKTxJ/IzMxErtD3J0r7drDyqjK2HaTaookIkOhRckw4ZGbSQyzBN7Ia7BRWpFwoUXIxJ84cwSdB/BPZCO2Z48Y4APP21NOo9vjzZeEtR/Yu15Vlz4xiTqgshm6D3wP+FJo/72Kw+1GNT9yZ7SDsYoURAl/oWARmlvWwNp7REaEzTKESIS7zHu8AiBHnoTIHrOUb+LDquWZCuR0EmKhUcDOeLMlBjMIQ8POzo5FBZVGwg3vL3oa9jM+56t1Z/p3HoSE2FgeZIzk4G0V5rwL3iaM4V0CwkUzeVvfANkpg65ojPLtG8/iALWrntxmmoDc2KrUzpVz8h9IoVsroyUHhqThYxgH4cJRZtPPFzofYyMOi0KW9ihk6AXpMcnJaTnP8XCN6fx6WkbxC7imS59FxkdDhSRgo7OgYn510OcaeJCZZOSN8XX5QCY1fDeaWwahmH3f2WTlIbb2rmSbtgaH0nA/Hs4wb1+h88IoT2aZmQLQwOnImzqTNMxhT4rAsuDhJ4Pn0sWB+zbLcSSY/9DqjfVZTtZNBhcQlQhJCkiK02QJvaIEinBGWGaxP8IOeYb0ZmOgn+lOPkmdCyE/5kxlpiiKkWogiZAHf6J4hYxTOCRnr4sFIXw8opd+Uy3qrriq1/IdWs67WWlWl8aGOTzxVuPTnS0r5rJX/9KmFH5rKmdqqN5RyQakVkBn75vjNm9RtbAa4eFJTapcIAsq5++ZNCEItKyclVWwgYWuXaj2ivHLyMd+Crs0C/KbUyq2TUiX/ayuvVBvNmtpSmo1KTS2pSr1YPptXo1IptfKV8uSYfRRUvnJ+rswtK+PwG4uxwPJPhUU4G0OxrgB1gXYctqXWapXaTA31vNq4hD4WLuegVMvN88UNF8unxXKxobag/ebjgEsMo1wpw2Q21JpSapWbpSXIx0bsVVkOulapQFdKSr3+KIFq8EltlSqVakspK6XLejGCdxhkvVEr5tlavFFpXFajeIzDNevVYr5YadaBGS/UOVDNck1V8h8YT+YrhXlgrUKzWirmFSA93wVtnSuN/IcQdKV62ap+UOoq62I1IGiC50XPGVnO62d1tVwIIhBQDRW48KQIVA7xylk+38pHCirI6du3E6DCJbgUxTzMbgvGH+gFFkImikBFSFh9uoVKtVE8F55Jq6ReqCXfeLLyak0FwufVer1SaxVUxo0AWveV9EasoJ40z47xERf+/XVcNwfU1sElSYjMxARh/fL8BC+bVGvFC6Su+gmYqzzVazYFud1Wo9LKZlonxXm87cMqJ5VmA4QEtEe5hUwyy3s+bLNcUGGma2oBlGUJ5LpSi8TZLLPBKqXiPwAUVZLH0crZWQ0oUrxQZ6qA3i20TpvlfGNeVwXMhVIr4vQHYYrVD5WyWqkDoaulyuW5WvZuvQDUgSym5bxR8pQqIz0qiUoQTaVcumwp0IMLFTVuiGnrhV9RQCFLHw0sk1pOwCMUqwZmjqaW0aoaZbhq1KBoE9ema2261qbrL226pq1SGdRcHSTR0/armbaljdci+7S2BE+wBN5czVH2GxcwkgLyh9jfCLQcMgRCvU+ZgtPUD1vDfFCVAih/YQg8G8AdjSi/YiMZ+O0AJ8mCBxhTSkKDLizwnGTBu+mUxOBIdwDLfQFXEL9FdNt1vZLk2LGTYvto4s3Ua3mkYgJvBPB//Ozt1lZsLlDoPuIUuHCGhGmaHmx4mMKgNz4AVUqF05JyxmB2Kp2veVEY3CF8POS3Uf+1WAUeAWkslebNeZTxP03/QOP/2JyvJ/1lJj0s6CL6sSimEYx+zIYRo6JIPyqiOM1/M72ZE++cjrg9Hp0Jhu8XucOBwFPoJr1ouOhc6I7ewefcxGIrCqr8mBb+sRsjz0XRx21FkKKLdMyPo6jP7o8zMdIe30gRFyf8+O9y2wlHz3QJrCt0XPAcJ/5y3nPg9tKC8/+Zmfs/2dTu/l/9/c/nPFj7HOOck979fDs0vLv2x+xxC0LFvsdxrNk43TmI/fxeeveb/+OHJHg1n1Vgd17fifMS4rIivpAmbm7G2M7x4aLzgwJNUuCBNpN+o+/X94d+UHri+V/x/si5ZupXACgDE81vY9H9v/3Z89/sN6H/0vL/F0nvhmIKCUyh6RyKmT5mj64eJpNOd0CHmhN6NFEb4UssTvhdFJ4EwxzPPsfxXgKB9hpby/OfJT1R/tkv7i3bxur3P1JpvP+3Pv//49P3zD/qgmXaeML872fXv//yIul75z+o45/x/tduen3/70VStFcubv/jLD9DG4/7f+lcZvr337Lp/d2/+vsPzylYzzHOOcn7vYdpf+1IkgK/lRt+tU7mT+TILCQZiJ2yeOmte7RkTQF+ojkU3wviPLdsZX57fAnovLipDQNiT/h0+ctRc0KcvBP4Bi6+uETm9xOvODGH19Zv2CuKuqkZZA5F+DNDPnl4Rd6Zx7oRXwZd6MGyMT7gFyo9mjwxNdAdOVgGS/NwvwJPNf1SgTU+EJAG+yquufcpO+UbD7bMf794bgQwjDswgUH0N/igZfBtgHjoXv3kbrzuvw0wyfOm2X8BINi7p13y919mEhhl3byxrmk8VqNdClLdI9OX+2PkzVQeZMRmLvwjWDgvcfT8Lzstlx79lepnMgAL9P9uKhcuS2Uzqb3sWv//OfQ/eyxv7OqGrNi2ducczRbkLcPgke2oUh7ef0RTM33g7zEt1OjBHxhd1lysZqXG+pBfXJAvdPrNv8SwsCk0D3X2pqjXu2Vsjnfk2n9kRdgdL/9+vkZG0r4L0uM96bLD+6EfYV3ZjghtzudbDjcgHpqc/wv3wnaFcCcWGxc2lAC5vZEEsp48kAB/ynQ4cu/4W5l/mNJdp3Vap3X6E6T/A8R/+EAAlgAA"`;
